pkg/api/types.go

```diff
@@ -59,6 +59,15 @@ type Config struct {
 	// BuilderBaseImageVersion provides optional version information about the builder base image.
 	BuilderBaseImageVersion string
 
+	// RuntimeImage specifies the image that will be a base for resulting image
+	// and will be used for running an application. By default, BuilderImage is
+	// used for building and running, but the latter may be overridden.
+	RuntimeImage string
+
+	// RuntimeArtifacts specifies a list source/destination folders that will
```

Comment:
> What if you want on the copy operation to rename?

It's not supported by the current implementation. User should rename a file in his `assemble` script.

Also, it's not clear how to handle destination in this case: how I can distinguish between a new file name and directory? For example, in case of "app-1.0-SNAPSHOT.jar" -> "app" how s2i can decide whether "app" is a directory for the artifact or it's new name?